django/db/models/sql/query.py

```diff
@@ -890,18 +897,25 @@ def join(self, join, reuse=None):
         The 'reuse' parameter can be either None which means all joins are
         reusable, or it can be a set containing the aliases that can be reused.
 
+        The 'force_reuse' parameter is the same type of 'reuse' but with difference
+        that it bypasses the 'reuse' paramter. It is used when computing.
+
         A join is always created as LOUTER if the lhs alias is LOUTER to make
         sure chains like t1 LOUTER t2 INNER t3 aren't generated. All new
         joins are created as LOUTER if the join is nullable.
         """
-        reuse = [a for a, j in self.alias_map.items()
-                 if (reuse is None or a in reuse) and j == join]
+        if force_reuse:
+            reuse = [a for a, j in self.alias_map.items()
+                     if a in force_reuse and j.equals(join, with_filtered_relation=False)]
+        else:
+            reuse = [a for a, j in self.alias_map.items()
+                     if (reuse is None or a in reuse) and j == join]
         if reuse:
             self.ref_alias(reuse[0])
             return reuse[0]
```

Comment:
Whilst you're editing this code, I think reusing (...) the name `reuse` from the argument to the actual list of aliases that are reusable is confusing. I would call it `reusable_aliases` or similar.

Reply:
makes sense